media-api/app/lib/ImagePersistenceReasons.scala

```diff
@@ -100,17 +100,22 @@ object HasLeases extends PersistenceReason {
   override val reason: String = "leases"
 }
 
-case class IsInPersistedCollection(persistedCollections: List[String]) extends PersistenceReason {
-  override def shouldPersist(image: Image): Boolean = {
-    // list of the first element of each collection's `path`, i.e all the root collections
-    val collectionPaths: List[String] = image.collections.flatMap(_.path.headOption)
-
-    // is image in at least one persisted collection?
-    (collectionPaths diff persistedCollections).length < collectionPaths.length
+case class IsInPersistedCollection(maybePersistOnlyTheseCollections: Option[List[String]]) extends PersistenceReason {
+  override def shouldPersist(image: Image): Boolean = maybePersistOnlyTheseCollections match {
+    case None =>
+      image.collections.nonEmpty
+    case Some(persistedCollections) =>
+      // list of the first element of each collection's `path`, i.e all the root collections
+      val collectionPaths: List[String] = image.collections.flatMap(_.path.headOption)
+      // is image in at least one persisted collection?
+      (collectionPaths diff persistedCollections).length < collectionPaths.length
```

Comment:
```suggestion
      // list of the first element of each collection's `path`, i.e all the root collections
      val collectionPaths: List[String] = image.collections.flatMap(_.path.headOption)
      // is image in at least one persisted collection?
      (collectionPaths intersect persistedCollections).nonEmpty
```
I guess this is the same but I don't really understand the previous condition - this version seems clearer to read to me (but it's always worked fine in the previous condition, so not important!)

Reply:
I like this, have implemented thank you